backend/cms/models/page.py

```diff
@@ -200,8 +203,64 @@ def available_languages(self):
                 }
         return available_languages
 
+    @property
+    def source_translation(self):
+        source_language_tree_node = self.page.region.language_tree_nodes.get(language=self.language).parent
+        if source_language_tree_node:
+            return self.page.get_translation(source_language_tree_node.code)
+        return None
+
+    @property
+    def latest_public_revision(self):
+        return self.page.page_translations.filter(
+            language=self.language,
+            public=True,
+            status=self.REVIEW_FINISHED,
+        ).first()
+
+    @property
+    def latest_major_revision(self):
+        return self.page.page_translations.filter(
+            language=self.language,
+            minor_edit=False,
+        ).first()
+
+    @property
+    def latest_major_public_revision(self):
+        return self.page.page_translations.filter(
+            language=self.language,
+            public=True,
+            status=self.REVIEW_FINISHED,
+            minor_edit=False,
+        ).first()
+
+    @property
+    def previous_revision(self):
+        version = self.version - 1
+        return self.page.page_translations.filter(
+            language=self.language,
+            version=version,
+        ).first()
+
+    @property
+    def is_outdated(self):
+        source_translation = self.source_translation
+        # If self.language is the root language, this translation can never be outdated
+        if not source_translation:
+            return False
+        # If the source translation is outdated, this translation can not be up to date
+        if source_translation.is_outdated:
+            return True
+        self_revision = self.latest_major_public_revision
+        source_revision = source_translation.latest_major_public_revision
+        # If on of the translations has no major public revision, it cannot be outdated
+        if not self_revision or not source_revision:
+            return False
+        return self_revision.last_updated < source_revision.last_updated
+
```

Comment:
I don't get the idea here. Maybe someone else has to have a look at that.

Reply:
I find this very easy to understand :-D

Reply:
this is because you like linux, open source and astronomy....

Reply:
Isn't this the target group of our repository? :telescope: